src/main/java/com/monumental/ExceptionHandler.java

```diff
@@ -0,0 +1,27 @@
+package com.monumental;
+
+import com.monumental.exceptions.ResourceNotFoundException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+
+/**
+ * This class is responsible for intercepting all exceptions and deciding what should happen with the HTTP response
+ * It also is able to log these exceptions, which is really helpful for debugging since Spring doesn't always print
+ * stacktraces for most exceptions
+ */
+@ControllerAdvice
+public class ExceptionHandler {
```

Comment:
I got fed up with not seeing exception traces again and brought this back. We need to work on the JSON output that it sends to the client because we're not getting useful error messages on the front end currently